cmd/eval.go

```diff
@@ -154,9 +155,15 @@ Set the output format with the --format flag.
 			return nil
 		},
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := eval(args, params); err != nil {
+			if code, err := eval(args, params); err != nil {
 				fmt.Fprintln(os.Stderr, err)
-				os.Exit(1)
+				if params.fail {
```

Comment:
[nit] I'd think this should keep the convention that when a functions returns a non-nil error, the other return arguments should not be used. Also, I think these conditionals could be simplified...maybe something like

```go
code, err := eval(..)
if err != nil {
    // print
    os.Exit(1)
} // no need for else, os.Exit stops execution
if params.fail {
    os.Exit(code)
}
```

On a related note, there's also a `RunE` (instead of `Run`) in cobra, which I think should handle error returns; it would allow us to `return err` instead of exiting-out here.

Reply:
Of course the convention should be kept. The problem here is that the exit code is different on errors then before, 2 instead of 1, but I could of course hard code that to 2 to avoid using the other parameter. Or it could be changed so 2 means undefined instead of error, which would allow you suggestion to be used easily. What do you think?

Reply:
@kichristensen I think that hardcoding the exit statuses here like @srenatus suggested would be fine. As far as the actual codes go, I felt like 0 for defined, 1 for undefined, and then 2 for unknown errors made the most sense -- but it's not a sticking point. As long as there's a clear way to differentiate between the three cases, I think we're good.